sklearn/inspection/_h_statistic.py

```diff
@@ -0,0 +1,318 @@
+            Numerator of pairwise H-squared statistic.
+            Useful to see which feature pair has strongest absolute interaction.
+            Take square-root to get values on the scale of the predictions.
+
+        denominator_pairwise : ndarray of shape (n_pairs, output_dim)
+            Denominator of pairwise H-squared statistic. Used for appropriate
+            normalization of H.
+
+    References
+    ----------
+    .. [FRI] :doi:`J. H. Friedman and B. E. Popescu,
+            "Predictive Learning via Rule Ensembles",
+            The Annals of Applied Statistics, 2(3), 916-954,
+            2008. <10.1214/07-AOAS148>`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.ensemble import HistGradientBoostingRegressor
+    >>> from sklearn.inspection import permutation_importance, h_statistic
+    >>> from sklearn.datasets import load_diabetes
+    >>> X, y = load_diabetes(return_X_y=True)
+    >>> est = HistGradientBoostingRegressor(max_iter=5, max_depth=2).fit(X, y)
+    >>> m = 3
+    >>> imp = permutation_importance(est, X, y, random_state=0)
+    >>> top_m = np.argsort(imp.importances_mean)[-m:]
+    >>> H = h_statistic(est, X=X, features=top_m, random_state=4)
+    >>> # {'feature_pairs': [(3, 8), (3, 2), (8, 2)],
+    >>> #  'h_squared_pairwise': array([[0.00985985],
+    >>> #         [0.00927104],
+    >>> #         [0.03439926]]),
+    >>> #  'numerator_pairwise': array([[ 1.2955532 ],
+    >>> #         [ 1.2419687 ],
+    >>> #         [11.13358385]]),
+    >>> #  'denominator_pairwise': array([[131.39690331],
+    >>> #         [133.96210997],
+    >>> #         [323.6576595 ]])}
+    >>> # Interpretation: For features (8, 2), 3.4% of the joint effect variability
+    >>> # comes from their interaction. These two features also have strongest absolute
+    >>> # interaction across pairs ("numerator_pairwise").
+    """
+    reduce_binary = False  # In binary classification, we only retain second class probs
+
+    check_is_fitted(estimator)
+
+    if is_regressor(estimator):
+        pred_fun = getattr(estimator, "predict", None)
+        if pred_fun is None:
+            raise ValueError("The regressor has no predict() method.")
+    elif is_classifier(estimator):
+        if isinstance(estimator.classes_[0], np.ndarray):
+            raise ValueError("Multiclass-multioutput estimators are not supported")
+        reduce_binary = len(estimator.classes_) == 2
+        pred_fun = getattr(estimator, "predict_proba", None)
+        if pred_fun is None:
+            raise ValueError("The classifier has no predict_proba() method.")
+    else:
+        raise ValueError("'estimator' must be a regressor or classifier.")
+
+    # Use check_array only on lists and other non-array-likes / sparse. Do not
+    # convert DataFrame into a NumPy array.
+    if not (hasattr(X, "__array__") or sparse.issparse(X)):
+        X = check_array(X, force_all_finite="allow-nan", dtype=object)
```

Comment:
Codecov complains that this is never triggered in a test.

Reply:
This check is from partial_dependence, but I can't find a corresponding test. Not sure how to test this.